### PR TITLE
Remove usage of @Test(expectedExceptions...) from TestFileSingleStreamSpillerFactory

### PR DESCRIPTION
**Pull request by Damans227**
Linked to `@Test(expectedExceptions...)` cleanup issue #6974

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/spiller/TestFileSingleStreamSpillerFactory.java [hunk 1]
```diff
@@ -143,24 +143,27 @@ private Page buildPage()
...
-        List<Path> spillPaths = ImmutableList.of(spillPath1.toPath(), spillPath2.toPath());
-        FileSingleStreamSpillerFactory spillerFactory = spillerFactoryFactory(spillPaths, 0.0);
-
-        spillerFactory.create(types, bytes -> {}, newSimpleAggregatedMemoryContext().newLocalMemoryContext("test"));
+        assertThatThrownBy(() -> {
+            List<Type> types = ImmutableList.of(BIGINT);
+            List<Path> spillPaths = ImmutableList.of(spillPath1.toPath(), spillPath2.toPath());
+            FileSingleStreamSpillerFactory spillerFactory = spillerFactoryFactory(spillPaths, 0.0);
+            spillerFactory.create(types, bytes -> {}, newSimpleAggregatedMemoryContext().newLocalMemoryContext("test"));
+        }).isInstanceOf(RuntimeException.class).hasMessage("No free or healthy space available for spill");
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/spiller/TestFileSingleStreamSpillerFactory.java:154]: I would change the style like this. 
```java
        List<Type> types = ImmutableList.of(BIGINT);
        List<Path> spillPaths = ImmutableList.of(spillPath1.toPath(), spillPath2.toPath());
        FileSingleStreamSpillerFactory spillerFactory = spillerFactoryFactory(spillPaths, 0.0);

        assertThatThrownBy(() -> spillerFactory.create(types, bytes -> {}, newSimpleAggregatedMemoryContext().newLocalMemoryContext("test")))
                .isInstanceOf(RuntimeException.class)
                .hasMessage("No free or healthy space available for spill");
```

Subsequent `throwIfNoSpillPaths` is also.

**Review by ebyhr — COMMENTED** [review]
Please fix the commit message as `Use assertThatThrownBy in TestFileSingleStreamSpillerFactory`

https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

File: core/trino-main/src/test/java/io/trino/spiller/TestFileSingleStreamSpillerFactory.java (hunk 1 above)
Comment (Damans227) [reply core/trino-main/src/test/java/io/trino/spiller/TestFileSingleStreamSpillerFactory.java]: This makes a lot more sense, thanks!

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**